Add Flow CR documentation (#252)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/flow.adoc

= Deploy Flow

:description: In this tutorial, you'll deploy a Flow application using Hazelcast Platform Operator for Kubernetes.

{description}

== Prerequisites

- Flow requires the Management Center application to run and be configured properly. The Management Center application should be configured to use FLow.

> 

Reviewer: Please mention FLOW MC image should be used here --> `docker.io/hazelcast/management-center-flow`

Author: I have added it to the example config with the following description

[source,yaml]
----
...
hazelcastClusters:
- address: flow <1>
  name: flow <2>
jvm:
  args:
  - "-Dhazelcast.mc.flow.addresses=http://233.252.0.158/" <3>
  - "-Dhazelcast.mc.internal.flowAddresses=http://flow:9021" <4>
...
----

<1> The service name of the Flow application. The same as the Flow CR name.
<2> Hazelcast cluster name. Should always be `flow`.
<3> Flow Ingress public address.
<4> The internal address of Flow in the format of `http://<service-name>:9021`. The service name is the same as the Flow CR name.

- To connect to the database you need create a `Secret` with the DB credentials:

[source,shell]
----
kubectl create secret generic flow-db-secret \
  --from-literal=database=flow \
  --from-literal=username=flow \
  --from-literal=password="strong-password"
----

In the created secret `username` and `password` are required, whereas `database` field is optional and will default to `flow` if not provided.

== Run Flow

To run Flow on Kubernetes using Hazelcast Platform Operator you need to create a Flow CR.

[source,yaml,subs="attributes+"]
----
include::ROOT:example$/flow.yaml[]
----

> Additional flow configuration can be accheived by providing additional parameters via `OPTIONS` environment variable.

Reviewer 2: ```suggestion
Additional flow configuration can be achieved by providing additional parameters using the `OPTIONS` environment variable.
```

= Deploy Flow

:description: In this tutorial, you'll deploy a Flow application using Hazelcast Platform Operator for Kubernetes.

{description}

See xref:flow:introduction:index.adoc[Flow] for more detailed information.

== Prerequisites

- Flow requires the Management Center application to run and be configured properly. The Management Center application should be configured to use FLow.

[source,yaml]
----
...
image: docker.io/hazelcast/management-center-flow <1>
...
hazelcastClusters:
- address: flow <2>
  name: flow <3>
jvm:
  args:

>   - "-Dhazelcast.mc.flow.addresses=http://233.252.0.158/" <4>

Reviewer 3: Not sure if you commonly use this elsewhere, but this ip in the examples looks slightly strange to me, it being ASM reserved makes it look super specific, but I presume it's just a placeholder, could perhaps replace with a placeholder FQDN instead?

Author: Well, I specifically used the IP range reserved for documentation purposes as a placeholder. In the description, I have explained that it should be publicly available one 😄

  - "-Dhazelcast.mc.internal.flowAddresses=http://flow:9021" <5>
...
----

<1> Management Center Flow image.
<2> The service name of the Flow application. The same as the Flow Custom Resource (CR) name.
<3> Hazelcast cluster name. Should always be `flow`.
<4> Flow Ingress public address.
<5> The internal address of Flow in the format of `http://<service-name>:9021`. The service name is the same as the Flow CR name.

> - To connect to the database you need create a `Secret` with the DB credentials:

Reviewer: The DB have to be Postgre so we should mention in here or at the beginning of the section. WDYT?

Author: I thought we didn't mention it on purpose, don't we?

Reviewer: Why?

If flow mentioned about it we can give reference to it also

Author: Done
